server/midddleware/forwarder.go

```diff
@@ -0,0 +1,88 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/mwitkow/grpc-proxy/proxy"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/types"
+	"github.com/tigrisdata/tigris/util"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+	"google.golang.org/protobuf/types/known/anypb"
+)
+
+var (
+	UserAgent = "tigris-forwarder/" + util.Version
+)
+
+var clients map[string]*grpc.ClientConn
+
+// TODO: Sweep unused connections periodically
+func getClient(ctx context.Context, origin string) (*grpc.ClientConn, error) {
+	if c, ok := clients[origin]; ok {
```

Comment:
Scope is global for the client map, you need to access it using lock as ForwardRequest will be called concurrently. Why not have a forwarder here? `type Forwarder struct {}` and have map inside it? Also, if we have explicit forwarder then it would be easy to have its own scope and metrics.

Reply:
Good catch. Yes, will synchronize.